app/src/main/java/com/duckduckgo/app/tabs/db/TabsDao.kt

```diff
@@ -168,4 +169,14 @@ abstract class TabsDao {
         title: String?,
         viewed: Boolean,
     )
+
+    @Transaction
+    open fun updateTabsOrder(from: Int, to: Int) {
+        if (from != to) {
+            val newTabs = tabs().swap(from, to)
+            newTabs.forEachIndexed { index, tabEntity ->
+                updateTab(tabEntity.copy(position = index))
```

Comment:
Since we know the positions of the swapped tabs, we can just update whose DB entries as all other positions are unaffected.

Reply:
Actually the way drag & drop animation works is that the other tabs get shifted and we need to updated their positions too. For example, say you have tabs [A, B, C, D]. You move A to the last position, but that doesn’t swap A and D in-place. Instead, B moves to the first place, C to second, etc.